src/inmanta/deploy/scheduler.py

```diff
@@ -634,6 +638,7 @@ async def report_resource_state(
         attribute_hash: str,
         status: ComplianceStatus,
         deployment_result: Optional[DeploymentResult] = None,
+        blocked_status: Optional[BlockedStatus] = None,
```

Comment:
I think it would be cleaner to derive this from `deployment_result` here. Related to my comment on `handler.py`.